perma_web/api/resources.py

```diff
@@ -451,17 +453,35 @@ def obj_create(self, bundle, **kwargs):
                 content_type='image/png',
             ).save()
 
-            # kick off capture task
-            run_task(proxy_capture.s(link.guid, bundle.request.META.get('HTTP_USER_AGENT', '')))
+            try:
+                run_task(proxy_capture.s(link.guid, bundle.request.META.get('HTTP_USER_AGENT', '')))
+            except Exception as e:
+                print "Getting error:",e
 
         return bundle
 
     def obj_update(self, bundle, skip_errors=False, **kwargs):
-
+        is_private = bundle.obj.is_private
         bundle = super(LinkResource, self).obj_update(bundle, skip_errors, **kwargs)
 
         if bundle.data.get('folder', None):
             bundle.obj.move_to_folder_for_user(bundle.data['folder'], bundle.request.user)
+
+        if 'is_private' in bundle.data:
+            today = datetime.today()
+            newest_date = today - timedelta(days=1)
+            newest_date = timezone.make_aware(newest_date)
+            if bundle.obj.creation_timestamp < newest_date:
```

Comment:
I think the previous four lines can be done as `if bundle.obj.archive_timestamp() < timezone.now()`, which also solves the problem that this is just comparing date rather than time. See `LinkUser.can_delete()` for an example.

Actually to make this even more readable and DRY, could we move that comparison to a `Link.is_archive_eligible()` function (or some similar name that makes more sense), and call it both from this line and from `LinkUser.can_delete()`?